lib/rules/no-tracked-properties-from-args.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+const ERROR_MESSAGE =
+  'Do not use this.args to create @tracked properties as this will not be updated if the args change. Instead use a getter.';
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+/** @type {import('eslint').Rule.RuleModule} */
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'disallow creating @tracked properties from this.args',
+      category: 'Ember Decorator',
```

Comment:
Let's just put this under `Ember Octane` with the other decorator-related rules.